src/main/java/rx/Observable.java

```diff
@@ -1258,6 +1258,7 @@ public Completable toCompletable() {
         final static Observable<Object> INSTANCE = create(new OnSubscribe<Object>() {
             @Override
             public void call(Subscriber<? super Object> subscriber) {
+                subscriber.setProducer(ProducerHolder.INSTANCE);
```

Comment:
My unit test didn't work because I don't create a producer until setProducer is called called. If setProducer is optional then I'll probably have to refactor to use the producer arbiter.